WordPress/Classes/ViewRelated/Jetpack/Jetpack Restore/Restore Status/Coordinators/JetpackBackupStatusCoordinator.swift

```diff
@@ -12,56 +12,73 @@ class JetpackBackupStatusCoordinator {
 
+    private var timer: Timer?
+
     // MARK: - Init
 
     init(site: JetpackSiteRef,
+         downloadID: Int,
          view: JetpackBackupStatusView,
          service: JetpackBackupService? = nil,
          context: NSManagedObjectContext = ContextManager.sharedInstance().mainContext) {
         self.service = service ?? JetpackBackupService(managedObjectContext: context)
         self.site = site
+        self.downloadID = downloadID
         self.view = view
     }
 
     // MARK: - Public
 
-    func start() {
-        service.prepareBackup(for: site, success: { [weak self] backup in
-            self?.view.render(backup)
-            self?.pollBackupStatus(for: backup.downloadID)
-        }, failure: { [weak self] error in
-            DDLogError("Error preparing downloadable backup object: \(error.localizedDescription)")
+    func viewDidLoad() {
+        startPolling(for: downloadID)
+    }
 
-            self?.view.showError()
-        })
+    func viewWillDisappear() {
+        stopPolling()
     }
 
     // MARK: - Private
 
-    private func pollBackupStatus(for downloadID: Int) {
-        Timer.scheduledTimer(withTimeInterval: Constants.pollingInterval, repeats: true) { [weak self] timer in
-            guard let self = self else { return }
-            self.service.getBackupStatus(for: self.site, downloadID: downloadID, success: { backup in
+    private func startPolling(for downloadID: Int) {
+        guard timer == nil else {
+            return
+        }
 
-                // If a backup url exists, then we've finished creating a downloadable backup.
-                if backup.url != nil {
-                    timer.invalidate()
-                    self.view.showComplete(backup)
-                    return
-                }
+        timer = Timer.scheduledTimer(withTimeInterval: Constants.pollingInterval, repeats: true) { [weak self] timer in
+            self?.refreshBackupStatus(downloadID: downloadID)
+        }
+    }
 
-                self.view.render(backup)
+    private func stopPolling() {
+        timer?.invalidate()
+        timer = nil
+    }
 
-            }, failure: { [weak self] error in
-                DDLogError("Error fetching backup object: \(error.localizedDescription)")
+    private func refreshBackupStatus(downloadID: Int) {
+        service.getBackupStatus(for: self.site, downloadID: downloadID, success: { [weak self] backup in
+            guard let self = self else {
+                return
+            }
 
-                timer.invalidate()
-                self?.view.showError()
-            })
-        }
+            // If a backup url exists, then we've finished creating a downloadable backup.
+            if backup.url != nil {
+                self.view.showComplete(backup)
+                return
+            }
+
+            self.view.render(backup)
+
+        }, failure: { [weak self] error in
+            DDLogError("Error fetching backup object: \(error.localizedDescription)")
```

Comment:
I don't want to overcomplicate things (so take this with a grain of salt) but it might happen that just a single request fails and the backup is still ongoing?

In this case, we would show an error message while that is not _entirely_ true.

At the same time, adding logic for that feels like we're handling an edge case or something that the API should take care of... so not sure.

Reply:
> I don't want to overcomplicate things (so take this with a grain of salt) but it might happen that just a single request fails and the backup is still ongoing?

Ah, yeah. Currently it looks like we keep on retrying to fetch the status. Will address in a different PR / follow up with the Android team.

https://github.com/wordpress-mobile/WordPress-iOS/blob/0e3ae925fe783f6a9b901459f7a176ef9fc9c839/WordPress/Classes/Stores/ActivityStore.swift#L393-L425